src/methods/SANA.cpp

```diff
@@ -1106,9 +1106,37 @@ void SANA::performChange(int type) {
             item.second += localScoreSumIncChangeOp(localSimMatrixMap[item.first], source, oldTarget, newTarget);
     }
 
+
+
     double newCurrentScore = 0;
     bool makeChange = scoreComparison(newAligEdges, newInducedEdges, newTCSum, newLocalScoreSum, newWecSum, newNcSum, newCurrentScore, newEwecSum, newSquaredAligEdges, newExposedEdgesNumer);
+
+    if(wasBadMove && !makeChange)
+    {
+        #ifdef CORES
+			// Statistics on the emerging core alignment.
+			// only update pBad if it's nonzero; re-use previous nonzero pBad if the current one is zero.
+			static double pBad;
+			if(trueAcceptingProbability()>0) pBad = trueAcceptingProbability();
+			coreCount[source]++;
+			coreFreq[source][(*A)[source]]++;
+			weightedCoreFreq[source][(*A)[source]] += pBad;
+			totalCoreWeight[source] += pBad;
+		#endif
+    }
+
     if (makeChange) {
```

Comment:
Consider the case that it's a bad move but we're going to accept it anyway. Then the above (wasBadMove && !makeChange) is false and doesn't get executed. Then here, where the bad move is accepted, you're incrementing the bad guy's core count.